File: Data Structures & Algorithms/two-integer-sum-ii/submission-7.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int left = 0, right = numbers.length - 1;
        int[] result = new int[2];

        while(left < right) {
            int currentSum = numbers[left] + numbers[right];
            if(currentSum > target) {
                right--;
            }
            else if(currentSum < target) {
                left++;
            }
            else{
                result[0] = left + 1;
                result[1] = right + 1;
                left++;
                right--;
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-7.py
class Solution:
    def findMin(self, nums: List[int]) -> int:

        left, right = 0, len(nums) - 1

        while left < right:
            middle = left + (right - left) // 2
            if nums[middle+1] < nums[middle]:
                return nums[middle+1]
            elif nums[middle-1] > nums[middle]:
                return nums[middle]
            elif nums[right] > nums[middle]:
                right = middle - 1
            elif nums[right] < nums[middle]:
                left = middle + 1  
        
        return nums[left]